src/core_functions/aggregate/README.md

```diff
@@ -0,0 +1,221 @@
+# Aggregate Functions
+
+Aggregate functions combine a set of values into a single value.
+In DuckDB, they appear in several contexts:
+
+* As part of the `SELECT` list of a query with a `GROUP BY` clause (ordinary aggregation)
+* As the only elements of a query _without_ a `GROUP BY` clause (simple aggregation)
+* Modified by an `OVER` clause (windowed aggregation)
+* As an argument to the `list_aggregate` function (list aggregation)
+
+## Aggregation Operations
+
+In order to define an aggregate function, you need to define some operations.
+These operations accumulate data into a _state_ object that is specific to the aggregate.
+Unlike simple scalar functions, there are several of these:
```

Comment:
It becomes clear(er) later on. Still, maybe it would help to mention that this _state_ object can exist once for each group (ordinary aggregation), once (simple aggregation), ?? (windowed aggregation), and once for each row/list (list aggregation). Or maybe not be that specific here, but let the reader know there will be more details about the different operations later...